iptables/iptables.go

```diff
@@ -26,3 +32,9 @@ func AddRule(appPort, metadataAddress, hostInterface, hostIP string) error {
 
 	return nil
 }
+
+// CheckInterfaceExists - validates the interface passed exists for the given system
+func CheckInterfaceExists(hostInterface string) error {
+	_, err := net.InterfaceByName(hostInterface)
```

Comment:
So here is my example with using the interface as cbr0
![image](https://cloud.githubusercontent.com/assets/1334469/21240764/ae1a9dbe-c2da-11e6-8017-54224d173d14.png)

I traced that back to https://golang.org/src/net/interface.go line 155:

```
138 // InterfaceByName returns the interface specified by name.
139	func InterfaceByName(name string) (*Interface, error) {
140		if name == "" {
141			return nil, &OpError{Op: "route", Net: "ip+net", Source: nil, Addr: nil, Err: errInvalidInterfaceName}
142		}
143		ift, err := interfaceTable(0)
144		if err != nil {
145			return nil, &OpError{Op: "route", Net: "ip+net", Source: nil, Addr: nil, Err: err}
146		}
147		if len(ift) != 0 {
148			zoneCache.update(ift)
149		}
150		for _, ifi := range ift {
151			if name == ifi.Name {
152				return &ifi, nil
153			}
154		}
155		return nil, &OpError{Op: "route", Net: "ip+net", Source: nil, Addr: nil, Err: errNoSuchInterface}
156	}
```

It appears that sometimes the interfaces are not ready (or available)